opm/io/eclipse/OutputStream.hpp

```diff
@@ -357,6 +359,75 @@ namespace Opm { namespace EclIO { namespace OutputStream {
                        const std::vector<T>& data);
     };
 
+    class SummarySpecification
+    {
+    public:
+        using StartTime = std::chrono::system_clock::time_point;
+
+        enum class UnitConvention
+        {
+            Metric = 1,
+            Field  = 2,
+            Lab    = 3,
+            Pvt_M  = 4,
+        };
+
+        struct RestartSpecification
+        {
+            std::string root;
+            int step;
+        };
+
+        class Parameters
+        {
+        public:
+            void add(const std::string& keyword,
+                     const std::string& name,
+                     const int          num,
+                     const std::string& unit);
+
+            friend class SummarySpecification;
+
+        private:
+            std::vector<PaddedOutputString<8>> keywords{};
+            std::vector<PaddedOutputString<8>> names{};
```

Comment:
Maybe call it `wgnames` - as these are not general names but rather well and group names?

Reply:
Good idea.  I'll update.